### ISSUE TITLE
OAS 3+ has no property 'definitions'

### ISSUE THREAD

**Issue by frankcorneliusmartin** [flasgger/flasgger#267]
OAS 3.x is not valid as the property `definitions` (even though it is added as empty list) is automatically added by flasgger. In case OAS 3.x this has been replaced by components (see [here](https://swagger.io/blog/news/whats-new-in-openapi-3-0/)). I am not sure how to properly fix this as it has been used throughout the package. 

A dirty hack that works for me is simply removing the `definitions` from the dict. (https://github.com/IKNL/flasgger/commit/b4451780d801688f49e2a1405e00883502365412) 

I assumed (maybe wrong) that OAS3+ was supported as it was used in one of the [examples](https://github.com/rochacbruno/flasgger/blob/master/examples//use_openapi.py). Although no were else in the documentation it states that OAS3+ is supported.

**Comment by Deeuus** [comment]
Any news on this?

**Comment by Kiolali** [comment]
same problem here as well

**Comment by frankcorneliusmartin** [comment]
> Any news on this?
> 
> edit: Apparently you can define the schemas inside the template when you create the Swagger instance. Maybe update documentation to properly show this?

You can define `schemas` and `components` (and possibly others) in the template. However it is, as far as I know, not possible to remove the `definitions` label in the output-swagger. Which results in an invalid OAS 3.0 specification.

**Comment by Kiolali** [comment]
yes.. how did you solve the problem? modifying base.py in my virtual environment is not a solution for me.

**Comment by frankcorneliusmartin** [comment]
> yes.. how did you solve the problem? modifying base.py in my virtual environment is not a solution for me.

So we removed the part where `definitions` is added, see: https://github.com/IKNL/flasgger/commit/b4451780d801688f49e2a1405e00883502365412

We also published this hack on pypi, see link above.

**Comment by Kiolali** [comment]
ah it's a fork! Thanks!

**Comment by nachiketrss** [comment]
@Kiolali, @frankcorneliusmartin , @AndreasNesKile - I am using flasgger 0.9.4 - the generated swagger spec contains the "definitions" tag. Is there a fix to this yet?

**Comment by frankcorneliusmartin** [comment]
We made a hack: https://github.com/flasgger/flasgger/issues/267#issuecomment-494401828

**Comment by frankcorneliusmartin** [comment]
The `definitions` tag is still there but the UI no longer complains about it, it seems. So on that note ill close this issue.

**Comment by nachiketrss** [comment]
When pasted into the swagger editor (http://editor.swagger.io/), it generates validation error due to the definitions tag.

**Comment by frankcorneliusmartin** [comment]
Good point

**Comment by Kiolali** [comment]
It would be great if the issue could be solved. 
I still get errors if I try to generate Client stubs using the openapi-generator (Version 4.3.1 and 5.0.0-beta2): "attribute definitions is unexpected"

**Comment by mathieu1** [comment]
Another datapoint: the extraneous definitions key trips up https://github.com/p1c2u/openapi-spec-validator
